File: test/exception/FException.test.ts
import { FException } from "../../src/index";

import { assert } from "chai";

describe("FException test", function () {
	it("should NOT be instantable without inner errors", function () {
		let testEx: FException | undefined = undefined;
		try {
			throw new FException("Inner2");
		}
		catch (ex0) {
			try {
				throw new FException("Inner1", ex0 as FException);
			} catch (ex1) {
				try {
					throw new FException("Main", ex1 as FException);
				} catch (ex2) {
					testEx = ex2 as FException;
					// 	Console.WriteLine(ex2.Message);
					// Console.WriteLine(ex2.ToString());
				}
			}
		}

		assert.isDefined(testEx);
		assert.strictEqual(testEx.message, "Main");
		// TODO
	});
});

File: test/exception/f_exception_test.dart
// Copyright 2023, the Freemework.ORG project authors. Please see the AUTHORS
// file for details. All rights reserved. Use of this source code is governed
// by a BSD-style license that can be found in the LICENSE file.

import "dart:convert";

import "package:test/test.dart";

import "package:freemework/freemework.dart";

void main() {
  group("FException tests", () {
    setUp(() {});

    tearDown(() {});
    test("should NOT be instantable without inner errors", () {
      FException? testEx;
      try {
        throw FException("Inner2");
      } catch (ex0) {
        try {
          throw FException("Inner1", ex0 as FException);
        } catch (ex1) {
          try {
            throw FException("Main", ex1 as FException);
          } catch (ex2) {
            testEx = ex2 as FException;
          }
        }
      }

      expect(testEx, isNotNull);
      expect(testEx.message, equals("Main"));
      expect(testEx.innerException!.message, equals("Inner1"));
      expect(testEx.innerException!.innerException!.message, equals("Inner2"));
      expect(
          LineSplitter.split(testEx.toString()).first,
          equals(
            "FException: Main ---> FException: Inner1 ---> FException: Inner2",
          ));

      //
    });
  });
}
